core/core.go

```diff
@@ -570,7 +569,7 @@ func (app *App) GetOrders(page, perPage int, snapshotID string) (*rpc.GetOrdersR
 		app.muIdToSnapshotInfo.Unlock()
 	}
 
-	notRemovedFilter := app.db.Orders.IsRemovedIndex.ValueFilter([]byte{0})
+	notRemovedFilter := app.orderSelector.db.Orders.IsRemovedIndex.ValueFilter([]byte{0})
```

Comment:
It doesn't seem right to have `app` constantly reach into the `orderSelector` field in order to access `db`. Maybe we should re-add the `db` field to `app` as well? `db` is just a pointer so it's not expensive to have multiple copies of it.

Alternatively, `orderSelector` could own all queries and `app` doesn't access `orderSelector.db` directly. This would involve a bit of work creating helper methods, so I'm more inclined to go with the first option.